src/NzbDrone.Core/Download/Clients/Flood/Flood.cs

```diff
@@ -0,0 +1,231 @@
+using NzbDrone.Common.Http;
+using NzbDrone.Core.Configuration;
+using NzbDrone.Core.Download.Clients.Flood.Models;
+using NzbDrone.Core.MediaFiles.TorrentInfo;
+using NzbDrone.Core.Parser.Model;
+using NzbDrone.Core.RemotePathMappings;
+
+namespace NzbDrone.Core.Download.Clients.Flood
+{
+    public class Flood : TorrentClientBase<FloodSettings>
+    {
+        private readonly IFloodProxy _proxy;
+
+        public Flood(IFloodProxy proxy,
+                        ITorrentFileInfoReader torrentFileInfoReader,
+                        IHttpClient httpClient,
+                        IConfigService configService,
+                        IDiskProvider diskProvider,
+                        IRemotePathMappingService remotePathMappingService,
+                        Logger logger)
+            : base(torrentFileInfoReader, httpClient, configService, diskProvider, remotePathMappingService, logger)
+        {
+            _proxy = proxy;
+        }
+
+        private static IEnumerable<string> HandleTags(RemoteEpisode remoteEpisode, FloodSettings settings)
+        {
+            var result = new HashSet<string>();
+
+            if (settings.Tags.Any())
+            {
+                result.UnionWith(settings.Tags);
+            }
+
+            if (settings.AdditionalTags.Any())
+            {
+                foreach (var additionalTag in settings.AdditionalTags)
+                {
+                    switch (additionalTag)
+                    {
+                        case (int)AdditionalTags.TitleSlug:
+                            result.Add(remoteEpisode.Series.TitleSlug);
+                            break;
+                        case (int)AdditionalTags.Quality:
+                            result.Add(remoteEpisode.ParsedEpisodeInfo.Quality.Quality.ToString());
+                            break;
+                        case (int)AdditionalTags.Language:
+                            result.Add(remoteEpisode.ParsedEpisodeInfo.Language.ToString());
+                            break;
+                        case (int)AdditionalTags.ReleaseGroup:
+                            result.Add(remoteEpisode.ParsedEpisodeInfo.ReleaseGroup);
+                            break;
+                        case (int)AdditionalTags.Year:
+                            result.Add(remoteEpisode.Series.Year.ToString());
+                            break;
+                        case (int)AdditionalTags.Indexer:
+                            result.Add(remoteEpisode.Release.Indexer);
+                            break;
+                        case (int)AdditionalTags.Network:
+                            result.Add(remoteEpisode.Series.Network);
+                            break;
+                        default:
+                            throw new DownloadClientException("Unexpected additional tag ID");
+                    }
+                }
+            }
+
+            return result;
+        }
+
+        public override string Name => "Flood";
+
```

Comment:
Until we have remove support we need to add:
```
public override ProviderMessage Message => new ProviderMessage("Sonarr is unable to remove torrents that have finished seeding when using Flood", ProviderMessageType.Warning);
```

Reply:
Actually the current behavior is that Sonarr defaults to instant removal of completed torrents, which is not a desired behavior in BitTorrent community. 

If we want to add the warning, I think it would be a good idea to make `item.CanMoveFiles` and `item.CanBeRemoved` always false so Sonarr won't instant remove completed torrents. Otherwise the warning can be misleading and needs to be reworded.

Reply:
Yeah you’re absolutely correct, I missed that part. That + the warning then, please.